src/ray/protobuf/experimental/autoscaler.proto

```diff
@@ -82,44 +80,50 @@ message ClusterResourceConstraint {
   repeated ResourceRequest min_bundles = 1;
 }
 
+// Node status for a ray node.
+enum NodeStatus {
+  // Node is alive and running workloads.
+  RUNNING = 0;
+  // Node is dead.
+  DEAD = 1;
+  // Node is idle.
+  IDLE = 2;
+}
+
 message NodeState {
-  enum NodeStatus {
-    // Node is alive.
-    ALIVE = 0;
-    // Node is dead.
-    DEAD = 1;
-    // Node is being drained.
-    DRAIN_PENDING = 2;
-    // Node is being drained.
-    DRAIN_FAILED = 3;
-    // Node is being drained.
-    DRAINING = 4;
-    // Node is already drained, and ready to be removed.
-    DRAINED = 5;
-  }
   // The node id internal to Ray in binary format.
-  bytes node_id = 11;
+  bytes node_id = 1;
 
   // The instance id that the node is running on.
   // This is passed in when the node is registered.
-  string instance_id = 12;
+  string instance_id = 2;
+
+  // The node type name, e.g. ray-head-node, matching `available_node_types`
+  // in the autoscaler config. See `ray/autoscaler/ray-schema.json`
+  // Should be set when a ray node is starting - and this will be empty
+  // if it's not set when starting the node.
+  string ray_node_type_name = 3;
```

Comment:
Do we have an idea about which environment variable we should use yet?

Reply:
cc @lanbochen-anyscale @brucez-anyscale

Reply:
we could potentially also use `--name` option that's currently there? cc @scv119